src/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,234 @@
+
+    walletCfg.connection = new Connection({
+      network: config.network,
+      servers: [config.server],
+      connectionTimeout: config.connectionTimeout,
+    });
+
+    let multisigDebugData = null;
+    if (params.multisig) {
+      const pubkeys = params.multisig.wordsArray.map(w => walletUtils.getMultiSigXPubFromWords(w));
+      walletCfg.multisig = {
+        pubkeys,
+        minSignatures: params.multisig.minSignatures
+      };
+      multisigDebugData = {
+        words: params.multisig.wordsArray,
+        total: pubkeys.length,
+        ...walletCfg.multisig
+      };
+    } else if (!params.words) {
+      walletCfg.seed = wallet.generateWalletWords();
+    }
+
+    const store = new MemoryStore();
+    Storage.setStore(store);
+    wallet.executeGenerateWallet(
+      walletCfg.seed,
+      '',
+      walletCfg.pinCode,
+      walletCfg.password,
+      false,
+      walletCfg.multisig
+    );
+
+    const addresses = [];
+    const addressIntervalStart = params.addressIntervalStart || 0;
+    const addressIntervalEnd = params.addressIntervalEnd || 22;
+    for (let i = addressIntervalStart; i < addressIntervalEnd; ++i) {
+      const addr = wallet.generateAddress(i);
+      addresses.push(addr.toString());
+    }
+    const timeEnd = Date.now().valueOf();
+    const timeDiff = timeEnd - timeStart;
+
+    const returnObject = {
+      duration: timeDiff,
+      isUsed: false,
+      words: walletCfg.seed,
+      addresses
+    };
+    if (params.multisig) {
+      returnObject.multisigDebugData = multisigDebugData;
+    }
+    return returnObject;
+  }
+
+  /**
+   * Reads a JSON file containing wallets and parses it
+   * @returns {Promise<any>}
+   * @throws SyntaxError
+   * @private
+   */
+  async _deserializeWalletsFile() {
+    const dataBuffer = await fs.readFile(this.WALLETS_FILENAME);
+    const strData = dataBuffer.toString();
+
+    try {
+      const jsonData = JSON.parse(strData);
+      return jsonData;
+    } catch (err) {
+      console.error('Corrupt wallets file');
+      throw err;
+    }
+  }
+
+  /**
+   * Writes the contents of a wallet array in a human-readable way, but with one wallet per line
+   * @param {unknown[]} wallets
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _serializeWalletsFile(wallets) {
+    /*
+     * The main aim of this file structure is human readability for debugging.
+     * The result must be a valid JSON, but with only one line per wallet.
+     */
+    let strWalletsData = wallets
+      .map(w => `${JSON.stringify(w)}`)
+      .join(',\n');
+    strWalletsData = `[\n${strWalletsData}\n]`;
+
+    await fs.writeFile(
+      this.WALLETS_FILENAME,
+      strWalletsData
+    );
+  }
+
+  /**
+   * Generates multiple new wallets and return them on an array
+   * @param [params]
+   * @param {number} [params.commonWallets=100] Amount of common wallets to be generated
+//   * @param {number} [params.multisigWallets=100] Amount of multisig wallets to be generated
```

Comment:
Should we uncomment this comment? or just remove it?

Reply:
I expected to implement an arbitrary amount of multisig wallets too.
This was transformed into the `generateMultisigWalletsForWords` method, which I believe is enough for this iteration of the tests.
Removed in [a06de8c](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/a06de8c399d6ef860b5378e420b61bbe7789be1e)